dashboard/app/controllers/api_controller.rb

```diff
@@ -243,16 +243,22 @@ def section_assessments
         # Don't allow somebody to peek inside an anonymous survey using this API.
         next if script_level.anonymous?
 
+        # Get the UserLevel for the last attempt.  This approach does not check
+        # for the script and so it'll find the student's attempt at this level for
+        # any script in which they have encountered that level.
         last_attempt = student.last_attempt_for_any(script_level.levels)
+
+        # Get the Level itself.
         level_group = last_attempt.try(:level) || script_level.oldest_active_level
+
+        # Get the response which will be stringified JSON.
         response = last_attempt.try(:level_source).try(:data)
 
         next unless response
 
+        # Parse the response string into an object.
```

Comment:
Though I appreciate the comment (not suggesting you remove it), the code is 100% as clear as the comment.